### PR TITLE
[TVMC] Add option to dump TIR code to file

### PR DESCRIPTION
**Pull request by lhutton1**
Currently dumps the TIR code after the final phase of lowering (phase 3 from https://github.com/apache/tvm/blob/665dd413bc85d14f7836324daf7cc0dd9281c85a/gallery/how_to/extend_tvm/low_level_custom_pass.py#L152) before codegen. This is done by running a pass to capture the TIR module as it is not saved during the build. The result is saved to file similar to the other code dumps.

In the future it seems good to have more granularity in the IR that is being dumped, e.g. print IR before/after named pass. I thought this might be helpful in the meantime.

cc @ekalda @ashutosh-arm @neildhickey

### CODE REVIEW

File: tests/python/driver/tvmc/test_compiler.py [hunk 1]
```diff
@@ -87,6 +88,16 @@ def test_compile_tflite_module(use_vm, tflite_mobilenet_v1_1_quant):
...
 
 
+def test_tir_dump(tflite_mobilenet_v1_1_quant):
+    pytest.importorskip("tflite")
+    tvmc_model = tvmc.load(tflite_mobilenet_v1_1_quant)
+    tvmc_package = tvmc.compile(tvmc_model, target="llvm", dump_code="tir")
+    dumps_path = tvmc_package.package_path + ".tir"
+    assert os.path.exists(dumps_path)
+    with open(dumps_path) as f:
+        assert "tir" in f.read()
```
Comment (ashutosh-arm) [inline tests/python/driver/tvmc/test_compiler.py:98]: @lhutton1, feels great to see this addition to `dump_code` :smile: Just a small question about testing: does this file already have a test for dumping multiple IRs from the same command?

Comment (lhutton1) [reply tests/python/driver/tvmc/test_compiler.py:98]: It does not unfortunately, I'll add one

Comment (lhutton1) [reply tests/python/driver/tvmc/test_compiler.py:98]: Added

**Review by ashutosh-arm — APPROVED** [review]
LGTM! Thank you :smile:

**Comment by neildhickey** [comment]
LGTM too. Thanks

**Review by echuraev — APPROVED** [review]
LGTM. Thanks

**Comment by ashutosh-arm** [comment]
Thanks @lhutton1, @neildhickey and @echuraev for the commits and the reviews :smile:

**Comment by PhilippvK** [comment]
> In the future it seems good to have more granularity in the IR that is being dumped, e.g. print IR before/after named pass. I thought this might be helpful in the meantime.

@lhutton1 I really like these changes. Are you going to do a followup for this soon? If not it would be great if we could open up an issue to collect some ideas. I plan to do something similar for the RelayIR dump (allow dumping post/pre-partitioning IR) and would like to follow a similar approach.

**Comment by lhutton1** [comment]
Hi @PhilippvK, thanks for the interest! I'm hoping to have a follow-up soon (~1wk) but I have another commitment I need to work on first. I'll summarise the thoughts I have in a discuss post - it would be good to see if they align to your thoughts/requirements before diving in

**Pull request merged**